Test/ApiFunctional/Model/Checkout/GetQuoteTest.php

```diff
@@ -0,0 +1,193 @@
+            'order_details' => [
+                'order_position' => 'product-detail-page',
+                'plugin_version' => 'Magento 1.0.32'
+            ],
+            'address_details' => [
+                'city' => 'Cairo',
+                'state' => 'Cairo',
+                'postcode' => '4472001',
+                'country' => 'EG',
+                'street1' => 'El-Thawra Street Sheraton Al Matar',
+                'street2' => '',
+                'email' => 'mardi@amwal.tech'
+            ],
+            'refId' => $tempData['refId'],
+            'uniqueRef' => false
+        ];
+
+        $transactionData = $this->_executeCurl('https://qa-backend.sa.amwal.tech/transactions/', $requestData);
+        $this->assertNotEmpty($transactionData);
+
+        $serviceInfoForGetQuote = [
+            'rest' => [
+                'resourcePath' => self::RESOURCE_PATH,
+                'httpMethod' => Request::HTTP_METHOD_POST,
+            ],
+            'soap' => [
+                'service' => self::SERVICE_NAME,
+                'serviceVersion' => self::SERVICE_VERSION,
+                'operation' => self::SERVICE_NAME . 'GetQuote',
+            ],
+        ];
+
+        $requestData = [
+            'ref_id' => $tempData['refId'],
+            'address_data' => [
+                'id' => '6e369835-451c-4071-8d86-496bd4a19eb6',
+                'street1' => '192 Nasr El Din, Haram, Giza, 12511',
+                'country' => 'SA',
+                'city' => 'Giza',
+                'state' => 'EG',
+                'postcode' => '12511',
+                'client_phone_number' => '+201033233462',
+                'client_email' => 'mardi@amwal.tech',
+                'client_first_name' => 'Ahmed',
+                'client_last_name' => 'Mardi',
+                'orderId' => $transactionData['id'],
+            ],
+            'is_pre_checkout' => false,
+            'trigger_context' => 'product-detail-page',
+            'ref_id_data' => [
+                'identifier' => '100',
+                'customer_id' => '0',
+                'timestamp' => '1707916143',
+            ],
+            'order_items' => [],
+            'cartId' => $tempData['cartId'],
+        ];
+
+        $response = $this->_webApiCall($serviceInfoForGetQuote, $requestData);
+        $this->assertIsArray($response);
+        $response = reset($response);
+
+        // Perform assertions
+        $this->assertArrayHasKey('cart_id', $response);
+        $this->assertArrayHasKey('available_rates', $response);
+        $this->assertArrayHasKey('amount', $response);
+        $this->assertArrayHasKey('subtotal', $response);
+        $this->assertArrayHasKey('tax_amount', $response);
+        $this->assertArrayHasKey('shipping_amount', $response);
+        $this->assertArrayHasKey('discount_amount', $response);
+        $this->assertArrayHasKey('additional_fee_amount', $response);
+        $this->assertArrayHasKey('additional_fee_description', $response);
```

Comment:
We can also introduce a `const` with required keys for this and use a foreach as mentioned in the other comment